gtfs-realtime/spec/en/reference.md

```diff
@@ -340,9 +354,89 @@ Identification information for the vehicle performing the trip.
 
 | _**Field Name**_ | _**Type**_ | _**Cardinality**_ | _**Description**_ |
 |------------------|------------|-------------------|-------------------|
-| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field |
+| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field. |
 | **label** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | User visible label, i.e., something that must be shown to the passenger to help identify the correct vehicle. |
 | **license_plate** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | The license plate of the vehicle. |
+| **carriage_descriptor** | [CarriageDescriptor](#message-carriagedescriptor) | repeated | An ordered list of carriage information. |
```

Comment:
@RachM Sorry for being late with reaction, I've been off-grid.

I'm still afraid your latest proposal wouldn't allow us to handle all the cases I'd like to see addressed.

Let me expand on the example of the Copenhagen-Karlskrona train. Before boring the Citytunneln in Malmö in 2010, the Malmö Central used to be a dead-end station as well. So that train had changed its direction twice during its journey (in Malmö and Kristianstad). I can't help myself, but I think to describe such situation we really have to add `CarriageDescriptor[]` as an optional field to `StopTimeUpdate` as well. Something like this:
`TripUpdate
    TripDescriptor
        CarriageDescriptor[]=[(A,B,C)+(D,E,F)] // two three-carriages units due to high demand
    StopTimeUpdate
        stop_id=Malmö, 
        CarriageDescriptor[]=[(F,E,D)+(C,B,A)] // first reversal
    StopTimeUpdate
        stop_id=Kristianstad, 
        CarriageDescriptor[]=[(A,B,C)] // second reversal, only single unit continues to Karlskrona
`
I found it very analogous to how `Trip.trip_headsign` can be overridden by `StopTime.stop_headsign`s in GTFS.

And on top of that it would be really great to have such info in GTFS too as it is quite common that the very same trip (train number) is served by quite different carriages on weekdays and on weekend.